service/lib/dinstaller/dbus/software.rb

```diff
@@ -39,9 +40,13 @@ def initialize(backend, logger)
         @backend = backend
         @logger = logger
 
+        @progress_handler = DInstaller::Progress.new
```

Comment:
We follow a different approach in `Manager`. The backend class `DInstaller::Manager` owns the progress object, and the D-Bus class configures the callbacks for such a progress object.